### PR TITLE
Add ProposalAttachmentType dynamic enum.

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

File: model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import lucuma.core.util.Enumerated
+import monocle.Focus
+import monocle.Lens
+
+case class ProposalAttachmentType(
+  tag:       String,
+  shortName: String,
+  longName:  String
+) derives Eq:
+  val accept: String = ".pdf" // only PDF files are currently allowed
```
Comment (rpiaggio) [inline model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala:20]: Can we have this as a constant on the object?

File: model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  val tag: Lens[ProposalAttachmentType, String]       = Focus[ProposalAttachmentType](_.tag)
+  val shortName: Lens[ProposalAttachmentType, String] = Focus[ProposalAttachmentType](_.shortName)
+  val longName: Lens[ProposalAttachmentType, String]  = Focus[ProposalAttachmentType](_.longName)
+
+  val values: List[ProposalAttachmentType] =
+    // This is a meta decoder, not a decoder for enum instances (which comes from the `Enumerated` instance)
+    given Decoder[ProposalAttachmentType] = semiauto.deriveDecoder
+
+    DynamicEnums.parsedEnums
```
Comment (rpiaggio) [inline model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala:33]: I wonder if we can abstract away the common logic of all dynamic enums.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala (hunk 1 above)
Comment (toddburnside) [reply model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala:20]: Yeah. We can. I was just anticipating the day when `they` decide that we need to support different file types for different attachment types, like we do for obs attachments. But, that's probably a bit too paranoid and easy to fix, anyway.

File: model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala (hunk 2 above)
Comment (toddburnside) [reply model/src/main/scala/lucuma/schemas/enums/ProposalAttachmentType.scala:33]: We might be able to. 🤔 I'll probably have to add `Partner` at some point soon. I'll look at it then.

**Pull request merged**